exist-core/src/test/java/org/exist/xquery/ModuleImportTest.java

```diff
@@ -0,0 +1,162 @@
+        } catch (final URISyntaxException e) {
+            fail(e);
+            return null;
+        }
+    }
+
+    @Test
+    public void importLibraryWithoutLocation() throws EXistException, PermissionDeniedException {
+        final Sequence expected = new StringValue("xs:integer");
+
+        final String query = "import module namespace functx='http://www.functx.com';" +
+                "functx:atomic-type(4)";
+        final Either<XPathException, Sequence> actual = executeQuery(query);
+
+        assertThatXQResult(actual, equalTo(expected));
+    }
+    @Test
+    public void importLibraryFromDbLocation() throws EXistException, PermissionDeniedException {
+        final Sequence expected = new StringValue("xs:integer");
+
+        final String query = "import module namespace functx='http://www.functx.com'" +
+                " at '/db/system/repo/functx-1.0.1/functx/functx.xq';" +
+                "functx:atomic-type(4)";
+        final Either<XPathException, Sequence> actual = executeQuery(query);
+
+        assertThatXQResult(actual, equalTo(expected));
+    }
+
+    @Test
+    public void importLibraryFromXMLDBLocation() throws EXistException, PermissionDeniedException {
+        final Sequence expected = new StringValue("xs:integer");
+
+        final String query = "import module namespace functx='http://www.functx.com'" +
+                " at 'xmldb:///db/system/repo/functx-1.0.1/functx/functx.xq';" +
+                "functx:atomic-type(4)";
+        final Either<XPathException, Sequence> actual = executeQuery(query);
+
+        assertThatXQResult(actual, equalTo(expected));
+    }
+
+    @Test
+    public void importLibraryFromExistXMLDBLocation() throws EXistException, PermissionDeniedException {
+        final Sequence expected = new StringValue("xs:integer");
+
+        final String query = "import module namespace functx='http://www.functx.com'" +
+                " at 'xmldb:exist:///db/system/repo/functx-1.0.1/functx/functx.xq';" +
+                "functx:atomic-type(4)";
+        final Either<XPathException, Sequence> actual = executeQuery(query);
+
+        assertThatXQResult(actual, equalTo(expected));
+    }
+
+    @Test
+    public void importLibraryFromUnknownLocation() throws EXistException, PermissionDeniedException {
+
+        final String query = "import module namespace functx='http://www.functx.com'" +
+                " at 'unknown:///db/system/repo/functx-1.0.1/functx/functx.xq';" +
+                "functx:atomic-type(4)";
+        final String expectedMessage = "error found while loading module functx: Source for module 'http://www.functx.com' not found module location hint URI 'unknown:///db/system/repo/functx-1.0.1/functx/functx.xq'.";
+
+        assertXQStaticError(ErrorCodes.XQST0059, -1,-1, expectedMessage, compileQuery(query));
+    }
+
+    @Test
+    public void importLibraryFromRelativeLocation() throws EXistException, PermissionDeniedException {
```

Comment:
this could help on the exist://  urls 

https://github.com/eXist-db/exist/blob/develop/exist-core/src/main/java/org/exist/protocolhandler/URLStreamHandlerStartupTrigger.java